pretyping/typing.ml

```diff
@@ -207,7 +213,11 @@ let check_allowed_sort env sigma ind c p =
   let pj = Retyping.get_judgment_of env sigma p in
   let _, s = hnf_decompose_prod env sigma pj.uj_type in
   let ksort = match EConstr.kind sigma s with
-  | Sort s -> ESorts.family sigma s
+  | Sort s ->
+    begin match ESorts.family sigma s with
+    | InType | InSProp | InSet | InProp as f -> f
+    | InQSort -> InType (* FIXME *)
```

Comment:
Fixme

Reply:
This is similar to #16797 and deferred for the same reason.

Reply:
It's not that similar IMO but w/e